app/src/org/commcare/utils/BiometricsHelper.java

```diff
@@ -151,11 +151,24 @@ private static int canAuthenticate(Context context, BiometricManager biometricMa
         return biometricManager.canAuthenticate(authenticator);
     }
 
-    private static void configureBiometric(Activity activity, int authenticator) {
+    private static boolean configureBiometric(Activity activity, int authenticator) {
         // Prompts the user to create credentials that your app accepts.
-        final Intent enrollIntent = new Intent(Settings.ACTION_BIOMETRIC_ENROLL);
-        enrollIntent.putExtra(Settings.EXTRA_BIOMETRIC_AUTHENTICATORS_ALLOWED,
-                authenticator);
+        Intent enrollIntent;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+            //Best case, handles both fingerprint and PIN
+            enrollIntent = new Intent(Settings.ACTION_BIOMETRIC_ENROLL);
+            enrollIntent.putExtra(Settings.EXTRA_BIOMETRIC_AUTHENTICATORS_ALLOWED,
+                    authenticator);
+        }
+        else if(authenticator == StrongBiometric && Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+            //An alternative for fingerprint enroll that might be available
+            enrollIntent = new Intent(Settings.ACTION_FINGERPRINT_ENROLL);
+        } else {
+            //No way to enroll, have to fail
```

Comment:
curious why did you decide to have a separate status enroll failed instead of directing user to settings screen from here ?

Reply:
Since in this case we're going to dump them on the Security settings screen and expect them to navigate manually from there (i.e. open Screen Lock setting, the configure desired biometrics), I'm navigating back up to where we can display the formal message screen so we can tell them what's about to happen and let them click the button to go on from there. Launching the ConnectIdMessageActivity from here seemed like unnecessarily breaking the pattern.